boa/src/builtins/array/mod.rs

```diff
@@ -414,25 +414,25 @@ impl Array {
         if len == 0 {
             this.set_field("length", 0);
```

Comment:
For my understanding - why do we set the field length to 0 (line 415) if we already know it is 0 from above (line 412)?

Reply:
I have not idea. its also in the spec... https://tc39.es/ecma262/#sec-array.prototype.shift.

Reply:
That is weird... hmm